tests/agent_unittests/test_package_version_utils.py

```diff
@@ -150,3 +150,21 @@ def test_version_caching(monkeypatch):
     del sys.modules["mymodule"]
     version = get_package_version("mymodule")
     assert version not in NULL_VERSIONS, version
+
+
+@SKIP_IF_NOT_IMPORTLIB_METADATA
```

Comment:
Instead of only running this against importlib (though I realize we are going to drop 3.7 soon in which case it would only use importlib) I think we should just do a more inclusive assertion at the bottom like the rest of the tests and not replace the importlib version-just let importlib or pkg_resources find the existing pytest version:
```
    assert version not in NULL_VERSIONS, version
```

Reply:
Could also add to this assert isinstance(version, str) or something to make sure it's not a property like it was before.